lib/postgrex/notifications.ex

```diff
@@ -121,10 +107,7 @@ defmodule Postgrex.Notifications do
   """
   @spec start_link(Keyword.t()) :: {:ok, pid} | {:error, Postgrex.Error.t() | term}
   def start_link(opts) do
-    {server_opts, opts} = Keyword.split(opts, [:name])
-    opts = Keyword.put_new(opts, :sync_connect, true)
-    connection_opts = Postgrex.Utils.default_opts(opts)
-    Connection.start_link(__MODULE__, connection_opts, server_opts)
+    SimpleConnection.start_link(__MODULE__, [], opts)
```

Comment:
```suggestion
    auto_reconnect = opts[:auto_reconnect] || false
    SimpleConnection.start_link(__MODULE__, [auto_reconnect: auto_reconnect], opts)
```

I think you need something like this and then to add `auto_reconnect` to the state so that on `handle_disconnect` you can reply with `{:eventually, ref}` if auto reconnect is true.

Reply:
Actually I don't think it's this simple because you only want `:eventually` on the listen command and not the regular queries.

Reply:
The listen command already issues an eventually reply when it isn’t connected. Without the auto-reply in error the API remains intact and the tests pass.

Reply:
I think the current refactor might be missing something important the old module was handling. But you guys let me know what you think.

If you look at the [listener_query](https://github.com/elixir-ecto/postgrex/blob/master/lib/postgrex/notifications.ex#L341) function in the old module, it is issuing the `:eventually` reply in 2 place: 

1. if the state is saying the socket is disconnected
2. if the state doesn't know the socket is disconnect and the user tries to issue a listen command and gets a disconnect message

I think the test with the auto-reply was exposing this gap between the old and new modules. i.e. this might happen:

1. Postgres server disconnects
2. Client issues listen command before the process knows the server disconnected (i.e. its state is not updated)
3. Client can't send eventually reply

Reply:
You’re right about the gap in functionality there being exposed by the failing test. The disconnect handler needs to reply with `:eventually` in this case, but should only do so if `auto_reconnect` is enabled.

Will fix 👍